src/test/java/org/example/homework13/BlackJackTesting/DeckTest.java

```diff
@@ -0,0 +1,56 @@
+package org.example.homework13.BlackJackTesting;
+
+import org.example.lesson14.blackjack.*;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+class DeckTest {
+
+    @Test
+    void drawCard() {
+        var deck = new Deck();
+        Card card = deck.drawCard();
+
+        assertEquals(51, deck.size(), "The deck should contain 51 cards after drawing a single card");
+        assertEquals(Rank.TWO, card.getRank(), "Rank of the drawing card should be the same as the transferred rank");
+        assertEquals(Suit.CLUBS, card.getSuit(), "Suit of the drawing card should be the same as the transferred suit");
+
+        for (int i = deck.size(); i > 0; i--) {
+            assertNotEquals(card, deck.drawCard(), "The first card should be missing from the deck");
+        }
+    }
+
+    @Test
+    void shuffle() {
+        var deck = new Deck();
+        var originalCards = deck.getCards();
+
+        deck.shuffle();
+
+        assertEquals(52, deck.size(), "The deck should contain 52 cards after the shuffle");
+
+        var shuffledCards = deck.getCards();
+        boolean arraysEqual = true;
+        int index = 0;
+
+        for (Card card : shuffledCards) {
```

Comment:
do not use for with collection where you can use streams of built-in collections methods
the whole test should be much shorter
```suggestion
    @Test
    void testShuffle() {
        List<Card> originalCards = new ArrayList<>(deck.getCards());
        deck.shuffle();
        List<Card> shuffledCards = deck.getCards();

        assertEquals(originalCards.size(), shuffledOrder.size(),
                "The size of the deck should remain unchanged");
        assertNotEquals(originalCards, shuffledOrder,
                "The deck should be shuffled and not in original order");
        assertTrue(shuffledOrder.containsAll(originalCards) && originalCards.containsAll(shuffledOrder),
                "The shuffled deck should contain all original cards");
    }
```